libraries/iio/iio.c

```diff
@@ -833,18 +858,31 @@ ssize_t iio_init(struct iio_desc **desc, struct iio_server_ops *iio_server_ops)
 	ops->close = iio_close_dev;
 	ops->get_mask = iio_get_mask;
 
-	ops->read = iio_server_ops->read;
-	ops->write = iio_server_ops->write;
+	ldesc->phy_type = init_param->phy_type;
+	if (init_param->phy_type == USE_UART) {
+		ret = uart_init((struct uart_desc **)&ldesc->phy_desc,
+					init_param->phy_init_param);
+		if (IS_ERR_VALUE(ret))
+			goto free_desc;
+		ops->read = iio_phy_read;
```

Comment:
You can move this outside if, since we already test `init_param->phy_type` inside `iio_phy_read`

Reply:
done